### ISSUE TITLE
Merge similar search results

### ISSUE THREAD

**Issue by Visne** [searxng/searxng#1502]
**Is your feature request related to a problem? Please describe.**
Sometimes, search results show two or more results which are essentially or exactly the same page on a website. I think the URL is always slightly different (but I'm not sure), but the title (so the text SearXNG shows) is the same. This is annoying, as two or more results linking to the same resource are shown in SearXNG. 

**Describe alternatives you've considered**
\-

**Additional context**
What I'm curious about is why SearXNG shows duplicate results while using Google, while Google does not show duplicates with the exact same query. Of course the easiest solution to this problem is to just use Google's (or another engine's) more accurately, although I'm not sure where it goes wrong.

**Comment by return42** [comment]
Duplicate of https://github.com/searxng/searxng/issues/1500 ?

**Comment by Visne** [comment]
> Duplicate of #1500 ?

No. By grouping pages I mean grouping pages which are not similar apart from the fact that they are on the same website.

This issue is about results which are extremely similar (such that they are essentially the same page, as far as the user is concerned) showing up next to eachother in the search results.

**Comment by Visne** [comment]
Here is an example I just stumbled upon:

![image](https://user-images.githubusercontent.com/39844191/178829597-065a4c8d-03a0-4e48-83c2-f462d8dccdec.png)

Clearly, the URLs are different. The `t` GET parameter is for threads, and the `p` parameter is for posts. The first result brings you to the post (top of the page), the second result brings you to a specific post (same page but scrolled down).

Meanwhile, Google's result is as follows:

![image](https://user-images.githubusercontent.com/39844191/178829943-a07b0da2-4542-42f6-b834-aaac02f3c2fa.png)

Google seems to only have the results with a `t` parameter. So then two questions arise:

1. Where does SearXNG get these semi-duplicate results, if they are not shown on Google's page?
2. How can they be filtered out?

**Comment by Visne** [comment]
Here's a bad case:

![image](https://user-images.githubusercontent.com/39844191/178844456-f62159ce-5306-4170-9e61-9b855d520283.png)

The only difference is the `l` get parameter, for language.

**Comment by Flepix** [comment]
In another engine, seems that Bing shows the same result in their own site https://www.reddit.com/r/bing/comments/cuta38/duplicate_search_results/

**Comment by return42** [comment]
> I think the URL is always slightly different (but I'm not sure)

The URL is different / SearXNG merges results with identical URLs.
